pipeline/src/transformers/addressables/event.ts

```diff
@@ -0,0 +1,73 @@
+import { asDate } from '@prismicio/client';
+
+import {
+  asText,
+  asTitle,
+  isFilledLinkToDocumentWithData,
+  isNotUndefined,
+} from '@weco/content-pipeline/src/helpers/type-guards';
+import { EventPrismicDocument } from '@weco/content-pipeline/src/types/prismic';
+import { ElasticsearchAddressableEvent } from '@weco/content-pipeline/src/types/transformed';
+
+export const transformAddressableEvent = (
+  document: EventPrismicDocument
+): ElasticsearchAddressableEvent | undefined => {
+  const { data, id, tags, uid: documentUid } = document;
+  const format = isFilledLinkToDocumentWithData(data.format)
+    ? asText(data.format.data.title)
+    : undefined;
+
+  const startEnd = data.times
+    .map(t => {
+      return {
+        start: asDate(t.startDateTime),
+        end: asDate(t.endDateTime),
+      };
+    })
+    .filter(d => d.start && d.end)
+    .sort((a, b) => Number(a.start) - Number(b.start));
+
+  const firstStart = startEnd?.at(0)?.start;
+  const lastEnd = startEnd?.at(-1)?.end;
+
+  const times = {
+    start: firstStart,
+    end: lastEnd,
+  };
+
+  const primaryImage = data.promo?.[0]?.primary;
+  const description = primaryImage?.caption && asText(primaryImage.caption);
+  const queryDescription = [description, format].filter(isNotUndefined);
+  const title = asTitle(data.title);
+  const uid = documentUid || undefined;
+  const contributors = (data.contributors ?? [])
+    .map(c => {
+      return isFilledLinkToDocumentWithData(c.contributor)
+        ? asText(c.contributor?.data.name)
+        : undefined;
+    })
+    .filter(isNotUndefined)
+    .join(', ');
+
+  return tags.includes('delist')
```

Comment:
This is speculative: rather than returning the transformed event with an [`isChildScheduledEvent`](https://github.com/wellcomecollection/content-api/blob/main/pipeline/src/transformers/eventDocument.ts#L171) boolean attached, we just return nothing, because we (probably?) don't want these to be indexed at all

Reply:
Opted against this approach and we'll bail out before these `delist`ed events reach this transformer stage

https://wellcome.slack.com/archives/C02ANCYL90E/p1732639950071859?thread_ts=1732638374.646629&cid=C02ANCYL90E